If new msg text is None, fallback to empty str

tehranian opened the pull request:
Equivalent change to b6ba473eabb6f622d24cca3be19efb013dde8481, but happening in a different place. See description from #138.

jtatum reviewed (changes requested):
Using both `.get()` and `or ''` seems redundant?

jtatum commented:
This seems to have come up a couple of times - if possible, can we add a unit or functional test for it? Let me know if you're comfortable doing that.

tehranian commented:
Hi @jtatum ,

re: `.get()` and `or ''` being redundant - Yea, I thought that too when I looked at the code. As described more verbosely in #138, the value under the key `text` is `None` so `.get()` returns `None`, not the fallback of empty-string. Subsequently, passing `None` into the regexp matcher raises an exception.

Here's an illustrative example:
```
>>> d = {'text': None}
>>> repr(d.get('text', ''))
'None'
>>> repr(d.get('text', '') or '')
"''"
```

re: Adding tests - Sure, will do.

tehranian committed:
Add unit test for #138

tehranian committed:
Defend against get_plugins() being called with None

tehranian commented:
Hi @jtatum,

Updated PR:
* Added unit test for #138, already merged.
* Moved logic of change for #144 to deeper-level, defending against bad input of `text==None`.
  * Added unit test for #144.

Testing Done:
```
================================================= 40 passed, 1 skipped in 97.21 seconds ==================================================
```

tehranian committed:
gitignore: .vscode

tehranian committed:
msg.get() in MessageDispatcher._dispatch_msg_handler()
